CMakeLists.txt

```diff
@@ -96,12 +96,20 @@ file(GLOB_RECURSE format_src
 
 # Exclude files from build directory to ensure they're not passed to the formatter
 list(FILTER format_src EXCLUDE REGEX "${CMAKE_CURRENT_BINARY_DIR}.")
+# Exclude generated files from formatting
+list(FILTER format_src EXCLUDE REGEX "${CMAKE_PROJECT_DIR}/include/.")
```

Comment:
I don't think we should be excluding the generated files. Instead, these files should be formatted immediately after being generating, so that they don't fail the `check-generated` check. Or, alternatively, we could manually fix the templates so that are correct upfront, but that might be a pita...

Reply:
Sounds good - I've made the generate target run `cppformat`